RichString.c

```diff
@@ -185,11 +185,9 @@ int RichString_findChar(const RichString* this, char c, int start) {
 
 #endif /* HAVE_LIBNCURSESW */
 
-void RichString_prune(RichString* this) {
+void RichString_delete(RichString* this) {
    if (this->chlen > RICHSTRING_MAXLEN)
       free(this->chptr);
-   memset(this, 0, sizeof(RichString));
-   this->chptr = this->chstr;
```

Comment:
Don#t drop this, because this leaves a dangling pointer to potentially unallocated memory in ´this-<chptr´